cmd/spiffe-aws-assume-role/cli/cli.go

```diff
@@ -39,7 +39,8 @@ func (c *CredentialsCmd) Run(context *CliContext) error {
 		c.RoleARN,
 		src,
 		c.SessionDuration,
-		context.STSProvider)
+		context.STSProvider,
+		c.STSEndpoint)
```

Comment:
nit: NewProvider is becoming a bit unwieldy with the number of arguments it's taking. It's probably fine for now, but I imagine we'd be passing some `AWSConfig` struct at some point instead.

Reply:
Totally agree. I think it makes sense to eliminate the STS provider in favor of just passing in the actual STS client and push the construction up a level. But I haven't done a solid analysis on exactly which other parameters can be rolled up and how.